### PR TITLE
Introduce FullClient

### PR DESCRIPTION
**Pull request by David-Petrov**
- refactor to separate the RpcApp from LightCLient
- extract bean stopping concerns to RpcApp::stop()
- introduce a FullClient alternative to the LightClient
- add logic in Main to switch client roles depending on the -node-mode CLI argument
- also, disable failing tests in SyncTest; there seems to be an issue with the chosen bootnode

### CODE REVIEW

File: src/main/java/com/limechain/Main.java [hunk 1]
```diff
@@ -1,15 +1,59 @@
...
+import lombok.extern.java.Log;
 import sun.misc.Signal;
 
+import java.util.logging.Level;
+
+@Log
 public class Main {
+
     public static void main(String[] args) {
+        // Instantiate and start the spring application, so we get teh global context
```
Comment (vikinatora) [inline src/main/java/com/limechain/Main.java]:
```suggestion
        // Instantiate and start the spring application, so we get the global context
```

**Review by vikinatora — APPROVED** [review]
LGTM, a lot of TODOs left but it's fine imo.

File: src/main/java/com/limechain/lightclient/FullClient.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.limechain.network.Network;
+import com.limechain.rpc.server.AppBean;
+import com.limechain.sync.warpsync.WarpSyncMachine;
+import lombok.SneakyThrows;
+import lombok.extern.java.Log;
+
+import java.util.logging.Level;
+
+@Log
+public class FullClient implements NodeClient {
```
Comment (vikinatora) [inline src/main/java/com/limechain/lightclient/FullClient.java]: FullNode is the term Polkadot uses.

File: src/main/java/com/limechain/lightclient/NodeClient.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.limechain.lightclient;
+
+// TODO: Think of a better name... that's tautological, simply `Client` is awfully generic
+public interface NodeClient {
```
Comment (vikinatora) [inline src/main/java/com/limechain/lightclient/NodeClient.java]:
```suggestion
// TODO: Think of a better name... that's tautological, simply `Client` is awfully generic
public interface HostNode {
```

**Review by georg-getz — APPROVED** [review]

File: src/main/java/com/limechain/lightclient/FullClient.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package com.limechain.lightclient;
```
Comment (David-Petrov) [inline src/main/java/com/limechain/lightclient/FullClient.java]: What about renaming the package from `lightclient` to something more fitting, like just `client`, or `node` or something... I'm not sure? @vikinatora

Comment (David-Petrov) [reply src/main/java/com/limechain/lightclient/FullClient.java:1]: @vikinatora

Comment (vikinatora) [reply src/main/java/com/limechain/lightclient/FullClient.java]: Makes sense 👍

Comment (David-Petrov) [reply src/main/java/com/limechain/lightclient/FullClient.java]: We'll go with `client`.

**Pull request merged**